pkg/build/strategies/sti/sti.go

```diff
@@ -196,6 +205,15 @@ func (builder *STI) Prepare(config *api.Config) error {
 		WorkingDir: config.WorkingDir,
 	}
 
+	if len(config.RuntimeImage) > 0 {
+		// For pulling RuntimeImage currently we're using the same PullPolicy as for BuilderImage
+		_, err := dockerpkg.PullImage(config.RuntimeImage, builder.docker, config.BuilderPullPolicy, false)
```

Comment:
the counter argument is that you don't want to do the build if the runtime image isn't available, so you should pull the runtime image first to make sure all the pre-reqs are available before you start building anything.  I think there's a decent argument for either ordering.

Starting the pull asynchronously and then waiting for it to complete (or checking it if it completed, and reporting any error) when we reach the point of needing it is a good idea, but i'll leave it up to @php-coder whether he thinks it would be too complicated to add that flow right now, it's definitely not a requirement.